src/main/java/fi/otavanopisto/restfulptv/server/services/ServiceEntityUpdater.java

```diff
@@ -93,16 +93,28 @@ private void startTimer(int duration) {
   public void onServiceIdUpdateRequest(@Observes ServiceIdUpdateRequest event) {
     if (!stopped) {
       if (event.isPriority()) {
-        queue.remove(event.getId());
-        queue.add(0, event.getId());
+        prependToQueue(event.getIds());
       } else {
-        if (!queue.contains(event.getId())) {
-          queue.add(event.getId());
-        }
+        appendToQueue(event.getIds());
       }
     }
   }
+  
+  private void prependToQueue(List<String> ids) {
+    for (String id : ids) {
+      queue.remove(id);
+      queue.add(0, id);
+    }
+  }
 
+  private void appendToQueue(List<String> ids) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "appendToQueue" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)